lib/pirates/game_server/instance.ex

```diff
@@ -98,10 +104,19 @@ defmodule Pirates.GameServer do
     {:reply, table, table}
   end
 
+  def handle_info(:tick, table) do
+    {pids, states} = Pirates.GameServer.Instance.state(table)
+    Enum.each(pids, fn pid -> send(pid, {:state_tick, states}) end)
+    {:noreply, table}
+  end
+
   # handle notifications of dead monitored processes
   def handle_info({:DOWN, _ref, :process, pid, _reason}, table) do
     :ets.delete(table, pid)
     IO.puts "Just deleted state for #{inspect(pid)}"
+    if count(table) == 0 do
+      exit(:shutdown)
```

Comment:
I think the _actual_ best practice is to return `{:stop, :shutdown, table}` here

> Returning `{:stop, reason, new_state}` stops the loop and terminate/2 is called with the reason reason and state new_state. The process exits with reason reason.

So maybe replace this and the next line with
```elixir
case count(table) do
  0 -> {:stop, :shutdown, table}
  _ -> {:noreply, table}
end